design/apps.go

```diff
@@ -0,0 +1,260 @@
+})
+
+var envStatMemory = a.Type("EnvStatMemory", func() {
+	a.Description(`memory stats`)
+	a.Attribute("used", d.Integer)
+	a.Attribute("quota", d.Integer)
+	a.Attribute("units", d.String)
+})
+
+var envStatPods = a.Type("EnvStatPods", func() {
+	a.Description(`pod stats`)
+	a.Attribute("starting", d.Integer)
+	a.Attribute("running", d.Integer)
+	a.Attribute("stopping", d.Integer)
+	a.Attribute("quota", d.Integer)
+})
+
+var timedIntTuple = a.Type("TimedIntTuple", func() {
+	a.Description("a set of time and integer values")
+	a.Attribute("time", d.Integer)
+	a.Attribute("value", d.Integer)
+})
+
+var simpleDeploymentStatSeries = a.Type("SimpleDeploymentStatSeries", func() {
+	a.Description("pod stat series")
+	a.Attribute("start", d.Integer)
+	a.Attribute("end", d.Integer)
+	a.Attribute("memory", a.ArrayOf(timedIntTuple))
+	a.Attribute("cores", a.ArrayOf(timedIntTuple))
+})
+
+var simpleSpaceSingle = JSONSingle(
+	"SimpleSpace", "Holds a single response to a space request",
+	simpleSpace,
+	nil)
+
+var simpleAppSingle = JSONSingle(
+	"SimpleApplication", "Holds a single response to a space/application request",
+	simpleApp,
+	nil)
+
+var simpleEnvironmentSingle = JSONSingle(
+	"SimpleEnvironment", "Holds a single response to a space/environment request",
+	simpleEnvironment,
+	nil)
+
+var simpleEnvironmentMultiple = JSONList(
+	"SimpleEnvironment", "Holds a response to a space/environment request",
+	simpleEnvironment,
+	nil,
+	nil)
+
+var simplePod = a.Type("SimplePod", func() {
+	a.Description("wrapper for a kubernetes Pod")
+	a.Attribute("pod", d.Any)
+})
+
+var simplePodMultiple = JSONList(
+	"SimplePod", "Holds a list of pods",
+	simplePod,
+	nil,
+	nil)
+
+var simpleDeploymentSingle = JSONSingle(
+	"SimpleDeployment", "Holds a single response to a space/application/deployment request",
+	simpleDeployment,
+	nil)
+
+var simpleDeploymentStatSeriesSingle = JSONSingle(
+	"SimpleDeploymentStatSeries", "HOlds a response to a stat series query",
+	simpleDeploymentStatSeries,
+	nil)
+
+var simpleEnvironmentStatSingle = JSONSingle(
+	"EnvStats", "Holds a single response to a pipeline/stats request",
+	envStats,
+	nil)
+
+var _ = a.Resource("apps", func() {
+	a.BasePath("/apps")
+
+	// An auth token is required to call the auth API to get an OpenShift auth token.
+	a.Security("jwt")
+
+	a.Action("showSpace", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID"),
+		)
+		a.Description("list applications in a space")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+		})
+		a.Response(d.OK, simpleSpaceSingle)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showSpaceApp", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/applications/:appName"),
+		)
+		a.Description("list application")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+		})
+		a.Response(d.OK, simpleAppSingle)
+		a.Response(d.NotFound, JSONAPIErrors)
```

Comment:
same comment as above about access restriction and `InternalServerError`

Reply:
see above comment